### PR TITLE
IDE-595 Background color for targets broken with new color preferences

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-595

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for your review. Found that the new language specific coloring broke the target background colors and that the target background color conflicted with a hard-coded gray background for ready-only files.

**Comment by GordonSmith** [comment]
@dehilsterlexis Should this target 6.2.x or 6.4.x?

File: EclEditor/LexGENERAL.cxx [hunk 1]
```diff
@@ -84,6 +84,7 @@ static void ColouriseGENERALDocSensitive(unsigned int startPos, int length, int
     WordList &hthorList = *keywordlists[2];
     WordList &roxieList = *keywordlists[3];
     WordList &localList = *keywordlists[4];
+    WordList &readyonlyList = *keywordlists[5];
```
Comment (GordonSmith) [inline EclEditor/LexGENERAL.cxx]: Should this be ```readonlyList```?

File: wlib/SourceView.cpp [hunk 2]
```diff
@@ -114,19 +114,21 @@ CSourceCtrl::CSourceCtrl(ISourceSlot * owner) : m_owner(owner), m_modified(false
 
 void CSourceCtrl::SetSourceType(const CString & typeStr)
 {
-    if (typeStr.Find(_T("roxie"),0) >= 0)
+    CString typeLower = typeStr;
+    typeLower = typeLower.MakeLower();
```
Comment (GordonSmith) [inline wlib/SourceView.cpp]: ```MakeLower``` lowers the text in place, so there is no need for the lhs assignment.

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you. I fixed on other small problem with Read Only which needed the space to be removed.

**Pull request merged**